src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs

```diff
@@ -82,30 +82,30 @@ internal static IVsProjectRestoreInfo Build(IEnumerable<IProjectVersionedValue<I
                     // https://github.com/dotnet/project-system/issues/3466for for details.
                     msbuildProjectExtensionsPath,
                     originalTargetFrameworks,
-                    new TargetFrameworks(targetFrameworks.Values),
-                    new ReferenceItems(toolReferences.Values)
+                    targetFrameworks.Values,
+                    toolReferences.Values
                 )
                 : null;
         }
 
-        private static IVsProjectProperties GetProperties(IImmutableDictionary<string, string> items)
+        private static IEnumerable<IVsProjectProperty> GetProperties(IImmutableDictionary<string, string> items)
         {
-            return new ProjectProperties(items.Select(v => new ProjectProperty(v.Key, v.Value)));
+            return items.Select(v => new ProjectProperty(v.Key, v.Value));
```

Comment:
I'm not sure why this is better?  It makes it unclear whether this is a deferred enumerator (as this is), and if that's okay.

Reply:
You and @drewnoakes had similar feedback, I've reverted back to using the strongly typed collections.